### ISSUE TITLE
assertFailedTransaction does not work

### ISSUE THREAD

**Issue by volodyad** [input-output-hk/plutus#3338]
Plutus, hash commit 6f834b5ca2276cb4bde0f25d8bf9518c3ff592b7

Test
 checkPredicate "Should fail if duplicate"
        ( 
            assertFailedTransaction (\_ err _ -> case err of {ScriptFailure (EvaluationError ["error message"]) -> True; _ -> False  })
        )
gives message - No transactions failed to validate, although in logs it shows error

Simulator logs 
      [INFO] Slot 13: W1: TxSubmit: cd86ded5d634b61cd573022a5c1494ac0d80f62a238ba1efe9469633056cbafc
      [INFO] Slot 13: TxnValidate cd86ded5d634b61cd573022a5c1494ac0d80f62a238ba1efe9469633056cbafc
      [WARNING] Slot 14: W1: handleTx failed: ValidationError (ScriptFailure (EvaluationError ["error message"]))

this assert worked well in older plutus starter version https://github.com/input-output-hk/plutus-starter/blob/main/cabal.project

**Comment by t1lde** [comment]
We ran into the same issue.

Some more info:

after bumping our pinned plutus version from 97437286b8df2bcafa665785a724eec0d2dcff7f to 5cdd2c3d708bf4c33514681dee096da6463273b7

`assertFailedTransaction` fails, with logs: 

```
Contract instance stopped with error: WalletError (ValidationError (ScriptFailure (EvaluationError ["..."])))
```

while `assertNoFailedTransactions` passes.

We have managed to workaround the issue with something like:

```haskell
assertContractError contract tag isScriptFailure "..."
  where
    isScriptFailure (WalletError (ValidationError (ScriptFailure _))) = True
    isScriptFailure _ = False
```

Some speculation on what could have caused this regression:
  - Perhaps https://github.com/input-output-hk/plutus/commit/86912cabdba4ffce534ac00c7ca24b4997465b3c affects some of the other folds?
  - Something to do with the Emulator / slot handling making the TxnValidationFail event not settle into the wallet emulator state?
  - Something to do with fee handling changing the way ChainEvents show up with failure?
